web/src/components/molecules/Common/Form/AssetItem/index.tsx

```diff
@@ -0,0 +1,125 @@
+import styled from "@emotion/styled";
+import { useState, Dispatch, SetStateAction } from "react";
+
+import Button from "@reearth-cms/components/atoms/Button";
+import Form, { FormItemProps, FormItemLabelProps } from "@reearth-cms/components/atoms/Form";
+import Icon from "@reearth-cms/components/atoms/Icon";
+import { UploadProps, UploadFile } from "@reearth-cms/components/atoms/Upload";
+import { Asset } from "@reearth-cms/components/molecules/Asset/asset.type";
+import { fileFormats, imageFormats } from "@reearth-cms/components/molecules/Common/Asset";
+import LinkToAssetModal from "@reearth-cms/components/molecules/Common/LinkToAssetModal/linkToAssetModal";
+import { useT } from "@reearth-cms/i18n";
+
+type Props = {
+  assetList: Asset[];
+  onAssetSearchTerm: (term?: string | undefined) => void;
+  onAssetsReload: () => void;
+  loadingAssets: boolean;
+  createAssets: (files: UploadFile[]) => Promise<void>;
+  fileList: UploadFile[];
+  setFileList: Dispatch<SetStateAction<UploadFile<File>[]>>;
+  setUploading: Dispatch<SetStateAction<boolean>>;
+  setUploadModalVisibility: Dispatch<SetStateAction<boolean>>;
+  uploading: boolean;
+  uploadModalVisibility: boolean;
+  onLink: (asset: Asset) => void;
+} & FormItemProps &
+  FormItemLabelProps;
+
+const AssetItem: React.FC<Props> = ({
+  name,
+  label,
+  extra,
+  rules,
+  assetList,
+  onAssetSearchTerm,
+  onAssetsReload,
+  loadingAssets,
+  createAssets,
+  fileList,
+  setFileList,
+  setUploading,
+  setUploadModalVisibility,
+  uploading,
+  uploadModalVisibility,
+  onLink,
+}) => {
+  const t = useT();
+  const { Item } = Form;
+
+  const [visible, setVisible] = useState(false);
+  const handleClick = () => {
+    setVisible(true);
+  };
+  const handleCancel = () => {
+    setVisible(false);
+  };
+  const displayUploadModal = () => {
+    setUploadModalVisibility(true);
+  };
+  const hideUploadModal = () => {
+    setUploadModalVisibility(false);
+    setUploading(false);
+    setFileList([]);
+  };
+
+  const handleUpload = () => {
+    setUploading(true);
+    createAssets(fileList).finally(() => {
+      hideUploadModal();
+      // TODO: link the uploaded asset with content after uploading is done
+    });
+  };
```

Comment:
useCallbacks

Reply:
Also can you use async/await. In most of our codebases we use async/await instead of `someFunction().finally()`

Reply:
Done!